proxy/api/src/http/keystore.rs

```diff
@@ -67,11 +63,13 @@ mod handler {
     }
 
     /// Initialize the keystore with a new key.
-    pub async fn create(mut ctx: context::Context) -> Result<impl Reply, Rejection> {
-        // TODO Load the real key from disk. The service manager ignores the key for now and uses a
-        // hardcoded one.
-        let key = coco::keys::SecretKey::new();
-        ctx.service_handle().set_secret_key(key);
+    pub async fn create(
+        mut ctx: context::Context,
+        input: super::CreateInput,
+    ) -> Result<impl Reply, Rejection> {
+        if let context::Context::Sealed(ref mut sealed) = ctx {
+            sealed.create_key(input.passphrase).await?;
+        }
```

Comment:
Does that mean on `NoSession`, it just returns `Ok` with the cookie set, but not actually create the keystorage?
I understood that when onboarding (ie no session and no keystore) the keystore then is created with a no-session-context.

Reply:
I’m not sure what you mean by `NoSession`. In this implementation we create a key in the keystore if the proxy is still sealed and no key is available. If the proxy is unsealed already we don’t create another key. We simply set the cookie again.

This is probably _not_ what we want though. Instead we should error with a 409 (conflict) or silently re-create the key. The first option creates some issue, though: Without adding a `DELETE /keystore` endpoint we can’t realize the flow of clearing the session and creating a second identity. I’ll fix it to use the second option.

Reply:
Fixed in https://github.com/radicle-dev/radicle-upstream/pull/1153/commits/e318d00817d752db4f837c0b4a834b9d4150fef2